ipscrub/src/ngx_ipscrub_support.c

```diff
@@ -36,3 +36,24 @@ ngx_int_t concat(ngx_pool_t *pool, ngx_str_t prefix, ngx_str_t suffix, u_char **
 
   return NGX_OK;
 }
+
+// randlong fills out with secure random bytes and returns NGX_OK iff successful.
+ngx_int_t randlong(long *out) {
```

Comment:
`long` is 64 bits on 64-bit architectures and 32 bits on 32-bit architectures. You need 128 bits of randomness at least to be secure, so you need to use another type (and also one that does not depend on the architecture, to ensure the same security level regardless), e.g. a `u_char[16]`.